gcp-common/src/main/java/io/micronaut/gcp/credentials/DefaultOAuth2HttpTransportFactory.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.credentials;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.LowLevelHttpRequest;
+import com.google.api.client.http.LowLevelHttpResponse;
+import com.google.api.client.util.StreamingContent;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.auth.oauth2.GoogleCredentials;
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.io.buffer.ByteBuffer;
+import io.micronaut.core.type.MutableHeaders;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.*;
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.HttpClientConfiguration;
+import io.micronaut.http.client.LoadBalancer;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An implementation of {@link HttpTransportFactory} based upon {@link HttpClient} that can be supplied
+ * when creating {@link GoogleCredentials}, allowing full control over configuration of the client.
+ *
+ * This transport will log unsuccessful HTTP requests at the {@code WARN} level using Micronaut framework's
+ * SLF4J logging infrastructure, giving more visibility into potential errors without needing to bridge the GCP
+ * SDK's java.util.logging statements.
+ *
+ * @author Jeremy Grelle
+ * @since 5.4.0
+ */
+@Singleton
+@Requires(classes = HttpClient.class)
+@Requires(property = GoogleCredentialsConfiguration.PREFIX + ".use-http-client", value = StringUtils.TRUE, defaultValue = StringUtils.TRUE)
```

Comment:
It misses test disabling the bean by setting `gcp.credentials.use-http-client` to `false`

Reply:
I added a test https://github.com/micronaut-projects/micronaut-gcp/pull/1015/commits/1da573bea4d3b295c2d535d7befb798318745067